packages/connections/src/components/connecting/connecting-animation.spec.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import ConnectingAnimation from './connecting-animation';
+
+describe('ConnectingAnimation Component', function () {
+  describe('when rendered', function () {
+    let unmountComponent;
+    let cancelAnimationFrameSpy;
+    let requestAnimationFrameSpy;
+    let timeoutCode;
+
+    beforeEach(function () {
+      if (!window.requestAnimationFrame) {
```

Comment:
`pretendToBeVisual` is a `jsdom` parameter. We're using the[ jsdom-global ](https://github.com/rstacruz/jsdom-global#readme)for our mocking of browser window apis, which doesn't have something like it. For now I've moved these functions into our register in the shared config. I think we could chat on switching to `jsdom` it looks more used and maintained.

Reply:
Hmmm, so `jsdom-global` uses `jsdom` internally and [seems to pass options through](https://github.com/rstacruz/jsdom-global/blob/371c36a2b64a7c2d014260212cc1a22a6bd1de2b/index.js#L29) to the jsdom. You are totally right though that it doesn't look like it was maintained in a while and it seems to have a hardcoded list of properties it will register in the global scope (so even have jsdom up to date doesn't guarantee that you'll get all the polyfills, IIRC @lerouxb ran into an issue with that in another package in the monorepo). Looks like [this `global-jsdom`](https://github.com/modosc/global-jsdom) seems to be in a better shape maintenance-wise and there seems to be enough setup code there so that we might want to just use the lib? We also can do the change only for the shared config, so no need to update it everywhere. Idk, anyway, it totally makes sense for me if you want to just have a local polyfill in the register file for now and open a code health ticket for later to adopt jsdom directly here or another `register` library

Reply:
Sounds good, updated the register.

Reply:
This is what I had to do in the end in that case https://github.com/mongodb-js/compass/pull/2502/commits/3a1f76ca6ceef9f7e5fbcbde8aa23f21c0758c21

Probably because the list of globals is hardcoded https://github.com/rstacruz/jsdom-global/blob/master/keys.js